internal/pkg/controller/controller.go

```diff
@@ -0,0 +1,163 @@
+package controller
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stakater/Reloader/pkg/kube"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	errorHandler "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+// ResourceUpdated contains new or updated objects
+type ResourceUpdated struct {
```

Comment:
Rename it to ResourceHandler
Move it to internal/pkg/handler/resourcehandler.go
Call handler.Handle() on the item

Have 2 resouce handlers
1. ResourceCreatedHandler
2. ResourceUpdatedHandler

Just log items in both their handle functions

Reply:
Names should be newResource and oldResource